src/app/account/mfa/sms/page.tsx

```diff
@@ -1,199 +1,186 @@
-                        <li key={phone.id} style={{ display: 'flex', gap: '10px' }}>
-                            <p>
-                                {phone.phoneNumber} {phone.defaultSecondFactor && '(Default)'}
-                            </p>
-                            <div>
-                                <Button onClick={() => phone.setReservedForSecondFactor({ reserved: false })}>
-                                    Disable for MFA
-                                </Button>
-                            </div>
-
-                            {!phone.defaultSecondFactor && (
-                                <div>
-                                    <Button onClick={() => phone.makeDefaultSecondFactor()}>Make default</Button>
-                                </div>
-                            )}
-
-                            {user.phoneNumbers.length > 1 && (
-                                <div>
-                                    <Button onClick={() => phone.destroy()}>Remove from account</Button>
-                                </div>
-                            )}
-                        </li>
-                    )
-                })}
-            </ul>
-            You have enabled MFA on your account
-            <Link href="/">
-                <Button>Return to homepage</Button>
-            </Link>
-        </>
-    )
-}
-
-// Display phone numbers that are not reserved for MFA
-const ManageAvailablePhoneNumbers = () => {
-    const { user } = useUser()
-
-    if (!user) return null
-
-    // Reserve a phone number for MFA
-    const reservePhoneForMfa = async (phone: PhoneNumberResource) => {
-        // Set the phone number as reserved for MFA
-        await phone.setReservedForSecondFactor({ reserved: true })
-        // Refresh the user information to reflect changes
-        await user.reload()
-    }
-
-    // phone numbers are valid for MFA but aren't used for it
-    const availableForMfaPhones = user.phoneNumbers.filter(
-        (phone) => phone.verification.status === 'verified' && !phone.reservedForSecondFactor,
-    )
+export default function ManageSMSMFA() {
+    const { isLoaded, user } = useUser()
+    const [phoneNumber, setPhoneNumber] = React.useState('')
+    const [verificationCode, setVerificationCode] = React.useState('')
+    const [error, setError] = React.useState('')
+    const [verificationSuccess, setVerificationSuccess] = React.useState(false)
+    const [codeSent, setCodeSent] = React.useState(false)
+    const [isSendingCode, setIsSendingCode] = React.useState(false)
+    const [isVerifying, setIsVerifying] = React.useState(false)
+    const [phoneResourceToVerify, setPhoneResourceToVerify] = React.useState<PhoneNumberResource | null>(null)
+
+    // Determine if there's an existing phone record
+    const existingPhone = user?.phoneNumbers?.[0]
+    const prefilledPhone = existingPhone ? existingPhone.phoneNumber : phoneNumber
+    const isPhoneEditable = !existingPhone
```

Comment:
it seems that at this point, if they already have an existing phone - they cant do anything in the form. why go through all the trouble of rendering all the disabled state if they are effectively locked out of using it? We can probably bail out and remove a lot of the logic around this later on in the component if we don't need to handle this case? 🤔

Reply:
@nathanstitt Yes, that idea was rejected during the process. I wasn’t aware that editing the Clerk profile screen was the preferred approach; otherwise, I likely would have voiced my objections to that design/UX decision.

Reply:
eh, I don't know if it was _preferred_ , just didn't know what we'd decided 😁 ty!